documentation/guide-jpa-query.asciidoc

```diff
@@ -52,6 +53,29 @@ Please be aware that code-generation can be painful especially with large teams.
   }
 ----
 
+.**quarkus**
+[source,java]
+----
+  public Page<Animal> findAllQueryDsl(AnimalSearchCriteriaDto dto) {
+
+    QAnimal animal = QAnimal.animal;
+    JPAQuery<Animal> query = new JPAQuery<Animal>(this.em);
+    query.from(animal);
+    if (dto.getName() != null && !dto.getName().isEmpty()) {
+      query.where(animal.name.eq(dto.getName()));
+    }
+    if (dto.getNumberOfLegs() != null) {
+      query.where(animal.numberOfLegs.eq(dto.getNumberOfLegs()));
+    }
+
+    // Order by name
+    query.orderBy(animal.name.desc());
+
+    List<Animal> animals = query.limit(dto.getPageSize()).offset(dto.getPageNumber() * dto.getPageSize()).fetch();
+    return new PageImpl<>(animals, PageRequest.of(dto.getPageNumber(), dto.getPageSize()), animals.size());
+  }
+----
+
```

Comment:
So to make it clear:
* show pagination only in according section below
* use QueryDSL generated code - we could copy the "old example" to a small separate devon4j-spring sepcific asciidoc file and link from here to avoid confusing users that started with devon4j and without generating these meta-classes.

Reply:
I have changed the code so that it is valid for both Spring and Quarkus applications using the QueryDSL metaclasses.  I also created a new file with the old example and linked it in the text.
Does the original code really come from a sample application? It looks like My-Thai-Star code, but I didn't find this code there. If not, maybe it would be better to provide a 'real' example.

Reply:
For the 'problem' regarding `Page` and `PageImpl` I created a new issue to check if there is a solution for Quarkus.
https://github.com/devonfw/devon4quarkus/issues/28